src/verify_cert.rs

```diff
@@ -1073,56 +1075,53 @@ mod tests {
 
         // Create a trust anchor, and use it to issue two distinct intermediate certificates, each
         // with a unique subject and keypair.
-        let trust_anchor = make_issuer("Trust Anchor");
+        let (trust_anchor, trust_anchor_cert) = make_issuer("Trust Anchor");
+        let trust_anchor_cert_der = trust_anchor_cert.der();
         let trust_anchor_cert =
-            Cert::from_der(untrusted::Input::from(trust_anchor.cert.der())).unwrap();
-        let trust_anchors = &[anchor_from_trusted_cert(trust_anchor.cert.der()).unwrap()];
+            Cert::from_der(untrusted::Input::from(trust_anchor_cert_der)).unwrap();
+        let trust_anchors = &[anchor_from_trusted_cert(trust_anchor_cert_der).unwrap()];
 
         let intermediate_a = issuer_params("Intermediate A");
         let intermediate_a_kp = KeyPair::generate_for(test_utils::RCGEN_SIGNATURE_ALG).unwrap();
-        let intermediate_a = intermediate_a
-            .signed_by(
-                &intermediate_a_kp,
-                &trust_anchor.cert,
-                &trust_anchor.key_pair,
-            )
+        let intermediate_a_cert = intermediate_a
+            .signed_by(&intermediate_a_kp, &trust_anchor)
             .unwrap();
+        let intermediate_a = Issuer::new(intermediate_a, intermediate_a_kp);
         let intermediate_a_cert =
-            Cert::from_der(untrusted::Input::from(intermediate_a.der())).unwrap();
+            Cert::from_der(untrusted::Input::from(intermediate_a_cert.der())).unwrap();
 
         let intermediate_c = issuer_params("Intermediate C");
         let intermediate_c_kp = KeyPair::generate_for(test_utils::RCGEN_SIGNATURE_ALG).unwrap();
-        let intermediate_c = intermediate_c
-            .signed_by(
-                &intermediate_c_kp,
-                &trust_anchor.cert,
-                &trust_anchor.key_pair,
-            )
+
+        let intermediate_c_cert = intermediate_c
+            .signed_by(&intermediate_c_kp, &trust_anchor)
             .unwrap();
+        let intermediate_c = Issuer::new(intermediate_c, intermediate_c_kp);
         let intermediate_c_cert =
-            Cert::from_der(untrusted::Input::from(intermediate_c.der())).unwrap();
+            Cert::from_der(untrusted::Input::from(intermediate_c_cert.der())).unwrap();
 
         // Next, create an intermediate that is issued by both of the intermediates above.
         // Both should share the same subject, and key pair, but will differ in the issuer.
         let intermediate_b_key = KeyPair::generate_for(test_utils::RCGEN_SIGNATURE_ALG).unwrap();
         let intermediate_b_params = issuer_params("Intermediate");
-        let intermediate_b_a = intermediate_b_params
+        let intermediate_b_a_cert = intermediate_b_params
```

Comment:
preexisting: finding this function pretty gnarly.

Reply:
Added two commits to slightly improve it. Can you be more specific about what aspects/parts you find gnarly?

Reply:
The large number of locals that have quite long identifiers but only differ by one character. This addresses most of that -- thanks!